### ISSUE TITLE
Is the "Do not use INFORMATION_SCHEMA views to determine the schema of an object." warning still true?

### ISSUE THREAD

**Issue by LarnuUK** [MicrosoftDocs/sql-docs#4188]
In the `INFORMATION_SCHEMA` documentation (such as this one), they all have a warning: 

> \*\* **Important** \*\* Do not use INFORMATION_SCHEMA views to determine the schema of an object. The only reliable way to find the schema of a object is to query the sys.objects catalog view.

Note that they specifically state that you should use the `sys.objects` view.

If you consult `sys.sp_helptext` however, you'll note that it is does indeed use `sys.objects`.

```sql
EXEC sys.sp_helptext @objname = N'INFORMATION_SCHEMA.COLUMNS';
```
And the relevant snippets from the output :
```sql
CREATE VIEW INFORMATION_SCHEMA.COLUMNS
AS
SELECT
	DB_NAME()									AS TABLE_CATALOG,
	SCHEMA_NAME(o.schema_id)						AS TABLE_SCHEMA, --important part
...
FROM
	sys.objects o JOIN sys.columns c ON c.object_id = o.object_id --Uses sys.objects
```
This output (or at least the relevant parts) are the same on SQL Server 2012-2019.

Does this mean that actual problem is the `SCHEMA_NAME()` function? [SCHEMA_NAME (Transact-SQL)](https://docs.microsoft.com/en-us/sql/t-sql/functions/schema-name-transact-sql?view=sql-server-ver15) contains no such warning if so.

This [issue](https://github.com/MicrosoftDocs/sql-docs/issues/2984) suggests this is still the case, but if so, why? If they are a problem, then does that actually mean that something else that `INFORMATION_SCHEMA` objects use is the problem?

#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 137b3d66-a524-b5a2-d5ab-752d91bea9ad
* Version Independent ID: c03d04b7-a72d-ad27-0035-affdfa3056ec
* Content: [COLUMNS (Transact-SQL) - SQL Server](https://docs.microsoft.com/en-us/sql/relational-databases/system-information-schema-views/columns-transact-sql?view=sql-server-ver15)
* Content Source: [docs/relational-databases/system-information-schema-views/columns-transact-sql.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/system-information-schema-views/columns-transact-sql.md)
* Product: **sql**
* Technology: **system-objects**
* GitHub Login: @CarlRabeler
* Microsoft Alias: **carlrab**

**Comment by CarlRabeler** [comment]
@LarnuUK  there are some cases where the information schema views are incomplete and that can lead to semantically broken applications if you are unaware

**Comment by LarnuUK** [comment]
@CarlRabeler I'm aware that there is a "known" problem, but the there's no information on the "why". The warning says *"to query the sys.objects catalog view", however, `INFORMATION_SCHEMA.COLUMNS` *uses* `sys.objects` according to `sp_helptext`. If `sys.objects` is correct, and the value of `SCHEMA_NAME` is wrong in `INFORMATION_SCHEMA.COLUMNS`, which is derived from `SCHEMA_NAME(o.schema_id)	` that **infers** the problem is `SCHEMA_NAME()`.

**Comment by CarlRabeler** [comment]
INFORMATION_SCHEMA  does not represent every possible state of the objects in SQL.  For example, filtered indexes.  The information_schema are incomplete.

**Comment by LarnuUK** [comment]
This isn't about the incompleteness of the objects, it's specifically about the warning about `TABLE_SCHEMA` having the possibility of being wrong, @CarlRabeler . As I've outlined.

**Comment by LarnuUK** [comment]
Can I *assume* that due to a lack of evidence to confirm that the problem with the `TABLE_SCHEMA` is true, and that as the objects *do* in fact use `sys.objects` means that the warning is no longer relevant to the `INFORMATION_SCHEMA` objects?

If so, then I'll happy check over the definitions of them (using `sp_helptext`) and then send a pull request on the relevant documents to remove the warning, where `SCHEMA_NAME(o.schema_id)	AS TABLE_SCHEMA` is used and the alias `o` refers to the catalog view `sys.objects`.

If the problem, however, is due to `TABLE_SCHEMA` then the warning should be included there (as there is no such warning on that function's documentation), and the warning on the `INFORMATION_SCHEMA` documentation amended to state that is the reason.

**Comment by CarlRabeler** [comment]
@LarnuUK if you check over the definitions and send a pull request, I will get it reviewed by the appropriate engineers. I have been trying to get a more complete answer, but it is still a work in progress.
Carl

**Comment by LarnuUK** [comment]
> 
> 
> @LarnuUK if you check over the definitions and send a pull request, I will get it reviewed by the appropriate engineers. I have been trying to get a more complete answer, but it is still a work in progress.
> Carl

If you're still waiting on an answer, @CarlRabeler  , that's fine; you hadn't mentioned you had reached out to them and things had gone a little quiet so wondered if a lack of evidence was effectively proof.

I appreciate that things like this can take awhile, and I would rather not create a pull request for a change that actually may end up removing important information (if for example `TABLE_SCHEMA` is the problem). I'm more than happy to wait for an "official" response if that's what this is waiting for.

Many thanks for reaching out to them.

**Comment by CarlRabeler** [comment]
@LarnuUK I cannot promise you I will get any more of a response than I have already replied to you with, but I have pushed it a bit - so far with no further response

**Comment by LarnuUK** [comment]
> 
> 
> @LarnuUK I cannot promise you I will get any more of a response than I have already replied to you with, but I have pushed it a bit - so far with no further response

Thanks, if we don't hear anything in the next month or so, I'll start working on a pull request; might get the cogs going instead. :)

Appreciate the attention you're giving this.

**Comment by CarlRabeler** [comment]
The more correct warning should be: "INFORMATION_SCHEMA views only represent a subset of the metadata of an object."

**Comment by CarlRabeler** [comment]
Do not use INFORMATION_SCHEMA views to determine the schema of an object. INFORMATION_SCHEMA views only represent a subset of the metadata of an object. The only reliable way to find the schema of a object is to query the sys.objects catalog view.
